### PR TITLE
add v1beta2 API

### PR DESCRIPTION
**Pull request by wanjunlei**
- Use trigger instead of input, now the function triggered by the triggers, and can get data from the inputs defined in `triggers`.
- Move the `.spec.serving.triggers` to `.spec.serving.scaleOptions.keda`.
- Move the `.spec.builder.params` to `.spec.builder.shipwright`.
- Add a new field `.spec.serving.workloadType` to determine which kind of workload will use.
- Now the function can use the existing dapr components.
- Add more information about the build and serving results to function status.
- Delete the `lastTransitionTime` field from the gateway status to prevent frequent triggering of reconcile.

### CODE REVIEW

**Comment by benjaminhuo** [comment]
Looks like something wrong with trigger
https://github.com/OpenFunction/OpenFunction/actions/runs/4979477134/jobs/8911123974?pr=442#step:16:66
image

**Comment by benjaminhuo** [comment]
> * Use trigger instead of input, now the function triggered by the triggers, and can get data from the inputs defined in `triggers`.
> * Move the `.spec.serving.triggers` to `.spec.serving.scaleOptions.keda`.
> * Move the `.spec.builder.params` to `.spec.builder.shipwright`.
> * Add a new field `.spec.serving.workloadType` to determine which kind of workload will use.
> * Now the function can use the existing dapr components.
> * Add more information about the build and serving results to function status.

The reconcile issue caused by the gateway status is fixed?

**Comment by wanjunlei** [comment]
> > * Use trigger instead of input, now the function triggered by the triggers, and can get data from the inputs defined in `triggers`.
> > * Move the `.spec.serving.triggers` to `.spec.serving.scaleOptions.keda`.
> > * Move the `.spec.builder.params` to `.spec.builder.shipwright`.
> > * Add a new field `.spec.serving.workloadType` to determine which kind of workload will use.
> > * Now the function can use the existing dapr components.
> > * Add more information about the build and serving results to function status.
> 
> The reconcile issue caused by the gateway status is fixed?

Yes

**Comment by wrongerror** [comment]
This function may need to be adjusted :
https://github.com/OpenFunction/OpenFunction/blob/main/pkg/core/serving/common/common.go#L578

**Comment by benjaminhuo** [comment]
> This function may need to be adjusted : https://github.com/OpenFunction/OpenFunction/blob/main/pkg/core/serving/common/common.go#L578

Good catch

**Comment by wanjunlei** [comment]
> > This function may need to be adjusted : https://github.com/OpenFunction/OpenFunction/blob/main/pkg/core/serving/common/common.go#L578
> 
> Good catch

It has been adjusted.

https://github.com/wanjunlei/OpenFunction/blob/main/pkg/core/serving/common/common.go#LL337C59-L337C59

**Review by wrongerror — APPROVED** [review]

File: apis/core/v1beta2/serving_types.go [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+	*DaprComponentRef `json:",inline"`
+	// Deprecated: Only for compatibility with v1beta1
+	InputName string `json:"inputName,omitempty"`
+}
+
+type DaprComponentRef struct {
+	// The name of the dapr component, the component can be defined in
+	// the `bindings`, `pubsub`, or `states`, or an existing component.
+	Name string `json:"name"`
+	// Type is the type of the component, if it is not set, controller will get it automatically.
```
Comment (benjaminhuo) [inline apis/core/v1beta2/serving_types.go:64]: We can add allowed type in comments

**Review by benjaminhuo — APPROVED** [review]

**Pull request merged**